packages/experimental/src/smart-session/utils/CosignerService.ts

```diff
@@ -38,26 +37,67 @@ export async function sendCoSignerRequest<
   transformRequest?: (data: TRequest) => unknown
 }): Promise<TResponse> {
   try {
+    // Transform request data if a transform function is provided
     const transformedData = transformRequest ? transformRequest(request) : request
-    const response = await axios.post<TResponse>(url, transformedData, {
-      params: queryParams,
-      headers
-    })
 
-    return response.data
+    // Construct query string
+    const queryString = new URLSearchParams({
+      ...queryParams
+    }).toString()
+
+    const fullUrl = queryString ? `${url}?${queryString}` : url
```

Comment:
Can we use the URL class to construct the fullUrl instead of manually concatenating strings? Using the URL class would ensure proper encoding of query parameters and make the code more robust and readable. For example:

```js
const fullUrl = new URL(url);
Object.entries(queryParams).forEach(([key, value]) => {
  fullUrl.searchParams.append(key, value);
});
```

This approach avoids potential issues with improperly encoded characters and handles edge cases gracefully.

@KannuSingh